lib/facility.rb

```diff
@@ -1,14 +1,66 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees, :engine, :year
+  attr_writer :registration_date, :plate_type
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") == true
+      collect_fee(vehicle)
```

Comment:
Nice job putting collect fee in its own method - promoting SRP here with helper functions